pkg/transfer/image/imagestore.go

```diff
@@ -212,9 +216,77 @@ func (is *Store) ImageFilter(h images.HandlerFunc, cs content.Store) images.Hand
 	return h
 }
 
-func (is *Store) Store(ctx context.Context, desc ocispec.Descriptor, store images.Store) ([]images.Image, error) {
+// checkUnpackAndAddPlatformLabel checks if the requested image has been successfully
+// unpacked for the requested platforms. If yes, platform image labels are added
+// to indicate what platforms the image has been pulled for. This becomes important
+// to support image pull per runtime class where an image could be pulled for different
+// platforms.
+// It also appends is.extraReferences with ImageID, repoDigest and repoTag as a bare minimum.
```

Comment:
Can we explore looking up here based on label rather using the extra references? Not using extra references allows doing this logic on `Create` or `Update` as well. Performing the logic on the image store operations allows wrapping the provided image store rather than expanding the interfaces. Long term we want the extra lookup and indexing to be done directly through an image store interface (higher level interface than images.Store today). Wrapping the provided store could make that easier.

Reply:
Sure I think that can be incorporated into main changes I wanted to bring out for image pull per runtime. The logic to look up based on id could happen using the label at CRI level and just the platform label to indicate what platform the image is being pulled for can be set here.

Reply:
Addressed this in the recent update, let me know if this looks good. Using the label to support lookup when an image could exist for multiple platforms will be added to the next PR that has the main implementation for image pull per runtime.